recipes/wayland/all/conanfile.py

```diff
@@ -79,22 +81,23 @@ def generate(self):
         tc.project_options["libraries"] = self.options.enable_libraries
         tc.project_options["dtd_validation"] = self.options.enable_dtd_validation
         tc.project_options["documentation"] = False
+        pc = PkgConfigDeps(self)
         if Version(self.version) >= "1.18.91":
             tc.project_options["scanner"] = True
+            if cross_building(self):
+                # Only wayland is the build_require. There is no need to apply suffixes.
+                pc.build_context_activated = ["wayland"]
+                tc.project_options["build.pkg_config_path"] = self.generators_folder
+            elif self.dependencies["expat"].is_build_context:  # wayland is being built as build_require
+                # If wayland is the build_require, all its dependencies are treated as build_requires
+                pc.build_context_activated = ["libffi", "expat", "libxml2", "libiconv", "zlib"]
```

Comment:
it cannot be hardcoded, the list of transitive dependencies will be different depending on the resolved graph.